src/core/lib/iomgr/tcp_posix.cc

```diff
@@ -574,16 +575,27 @@ static bool tcp_flush(grpc_exec_ctx* exec_ctx, grpc_tcp* tcp,
 
     if (sent_length < 0) {
       if (errno == EAGAIN) {
-        tcp->outgoing_slice_idx = unwind_slice_idx;
         tcp->outgoing_byte_idx = unwind_byte_idx;
+        // unref all and forget about all slices that have been written to this
+        // point
+        for (size_t idx = 0; idx < unwind_slice_idx; ++idx) {
```

Comment:
Should this consider whether the last slice has been written out partially? If that is the case, that slice should not be unreffed yet?

Reply:
I don't think so. In this case, we know sent_length < 0, so we know that nothing in the last write went out, so there should be no partial writing of a slice occurring. Plus this conditional will always return, so we won't make it to the logic that handles partial write of a slice.

Reply:
Let's say unwind_byte_idx != 0, which means the last slice has been sent out partially. It seems you are still unreffing that slice here.

Reply:
But unwind_byte_idx represents "byte within outgoing_buffer->slices[0] to write next"

So when this loop terminate, the slice at outgoing_buffer->slices[0] will be the one that has been partially written, and so next time we get into the flush function we will start from there.

Regardless, I will either ensure that tcp_posix_test is covering this case, or somehow add a test to verify this

Reply:
I think you are right. I forgot there is a increase again when entering the iteration.